File: Segundo período/AED-2/TP01-parte1/Arquivo.java
/*
 * Autor: Luiz Junio <luisjuniorbr@gmail.com>
 * Enunciado:
 *  Arquivo em Java: Faca um programa que leia um numero inteiro n indicando o numero de
 *  valores reais que devem ser lidos e salvos sequencialmente em um arquivo texto. Apos a leitura
 *  dos valores, devemos fechar o arquivo. Em seguida, reabri-lo e fazer a leitura de tras para frente
 *  usando os metodos getFilePointer e seek da classe RandomAccessFile e mostre todos os valores
 *  lidos na tela. Nessa questao, voce nao pode usar, arrays, strings ou qualquer estrutura de dados.
 *  A entrada padrao e composta por um numero inteiro n e mais n numeros reais. A saida padrao
 *  corresponde a n numeros reais mostrados um por linha de saıda.
 */

import java.io.RandomAccessFile;
import java.io.FileNotFoundException;
import java.io.IOException;

public class Arquivo
{//Inicio classe Arquivo
    public static void main(String[] args)
    {//Inicio main
        try
        {//Inicio try
            RandomAccessFile file = new RandomAccessFile("numeros.txt", "rw");
            int numEntrada = MyIO.readInt();//Ler quantidade de numeros que serão escritos no arquivo
            double valor;
            for(int i = 0; i < numEntrada; i++)
            {//Inicio for
                file.writeDouble(MyIO.readDouble());//Escreve o que foi digitado no arquivo
            }//Fim for

            file.close();//Fecha o arquivo

            file = new RandomAccessFile("numeros.txt", "r");//Abre o arquivo como leitura apenas

            for(int i = 0; i < numEntrada; i++)
            {//Inicio for
                file.seek(((numEntrada - 1 - i) * 8 ));//Entrada - 1 pq é de 0-79
                valor = file.readDouble();
                if(valor % 1 == 0)//Se for um numero inteiro, será impresso como inteiro
                    MyIO.println((int) valor);
                else
                    MyIO.println(valor);
            }//Fim for

            file.close();//Fecha o arquivo

        }//Fim try

        catch(FileNotFoundException fileNotFoundException){
            System.out.println(fileNotFoundException);
        }
        catch(IOException ioException){
            System.out.println(ioException);
        }

    }//Fim main 




}//Fim classe Arquivo

File: Segundo período/AED-2/TP01-parte1/Arquivo.c
/*
 * Autor: Luiz Junio <luisjuniorbr@gmail.com>
 * Criado em: 21/08/2018
 *
 *
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main()
{//Inicio main

    FILE *arquivo;
    int numEntrada = 0;
    double valor = 0.0;
    char string[30];
    char *tmp;

    if((arquivo = fopen("numeros.txt", "w")) == NULL ){
        printf("Arquivo não pode ser aberto\n");
    }
    else
    {//Inicio else
        scanf("%d", &numEntrada);
        for(int i = 0; i < numEntrada; i++){
            scanf("%s", string);
            valor = strtod(string, &tmp);
            fprintf(arquivo, "%lf\n", valor);
        }
        fclose(arquivo);

        if((arquivo = fopen("numeros.txt", "r")) == NULL){
            printf("Arquivo não pode ser aberto\n");
        }
        else{
            for(int i = numEntrada; i > -1; i--){ 
                fseek(arquivo, i * sizeof(double), SEEK_SET);
                fread(string, sizeof(double), 1, arquivo);
                //fscanf(arquivo ,"%s", string);
                valor = strtod(string , &tmp);//Converte String para double
                printf("%g\n", valor);
            }//Fim for

            fclose(arquivo);
        }//Fim else 2
    }//Fim else
    return 0;
}//Fim main 
